cmd/internal/cli/sign.go

```diff
@@ -110,36 +109,12 @@ var SignCmd = &cobra.Command{
 }
 
 func doSignCmd(cmd *cobra.Command, cpath string) {
-	// Group id should start at 1.
-	if cmd.Flag(verifySifGroupIDFlag.Name).Changed && sifGroupID == 0 {
-		sylog.Fatalf("invalid group id")
-	}
-
-	// Descriptor id should start at 1.
-	if cmd.Flag(verifySifDescSifIDFlag.Name).Changed && sifDescID == 0 {
-		sylog.Fatalf("invalid descriptor id")
-	}
-	if cmd.Flag(verifySifDescIDFlag.Name).Changed && sifDescID == 0 {
-		sylog.Fatalf("invalid descriptor id")
-	}
-
-	if sifGroupID != 0 && sifDescID != 0 {
-		sylog.Fatalf("only one of -i or -g may be set")
-	}
-
-	var isGroup bool
-	var id uint32
-	if sifGroupID != 0 {
-		isGroup = true
-		id = sifGroupID
-	} else {
-		id = sifDescID
-	}
-
-	if (id != 0 || isGroup) && signAll {
-		sylog.Fatalf("'--all' not compatible with '--sif-id' or '--groupid'")
+	id, isGroup, err := ensureImageFlags(cmd, cpath, sifDescID, sifGroupID, signAll)
+	if err != nil {
+		sylog.Fatalf("%s", err)
 	}
 
+	fmt.Printf("Signing image: %s\n", cpath)
```

Comment:
Do we ever need the `-s` silent flag to work on sign? I'm not sure if it is needed.

Reply:
The `-s` flag probably will not be very effective on `sign`, thats not something that will be fixed in this PR. And this `fmt.Printf(...)` was already there, I just moved it to a better location.